request.go

```diff
@@ -111,13 +112,18 @@ func (b *RequestBuilder) execute(req *http.Request) (Response, error) {
 	return Response{Request: b.request, RawResponse: response}, err
 }
 
-func (b *RequestBuilder) executeWithRetry(req *http.Request) (Response, error) {
+func (b *RequestBuilder) executeWithRetry() (Response, error) {
 	var config = b.request.config.RetryConfig()
 	var errExecution error
 	var errAttempts []error
 	var response Response
 
 	for attempt := uint(0); attempt < config.MaxAttempts(); attempt++ {
+		// For each unique attempt, create a new request instance to avoid side effects from previous attempts (e.g. request body)
+		req, err := b.createHTTPRequest()
```

Comment:
Using `b.createHTTPRequest` here doesn't resolve the problem. When creating a new `*http.Request` in `createHTTPRequest` we pass the `b.request.config.Body()` reference, which is an `io.Reader`.

We need to provide an alternative to reuse. Like a buffer:

```go
buffer := &bytes.Buffer{}
_, err = io.Copy(buffer, reader)
```

Reply:
yes, you're completely right. I made some changes and added two test cases to check if body keeps with the same size after the retries.